plugins/inventory/aws_ec2.py

```diff
@@ -577,16 +581,25 @@ def _add_hosts(self, hosts, group, hostnames):
             # Create groups based on variable values and add the corresponding hosts to it
             self._add_host_to_keyed_groups(self.get_option('keyed_groups'), host, hostname, strict=strict)
 
-    def _set_credentials(self):
+    def _set_credentials(self, loader):
         '''
             :param config_data: contents of the inventory config file
         '''
 
-        self.boto_profile = self.get_option('aws_profile')
-        self.aws_access_key_id = self.get_option('aws_access_key')
-        self.aws_secret_access_key = self.get_option('aws_secret_key')
-        self.aws_security_token = self.get_option('aws_security_token')
-        self.iam_role_arn = self.get_option('iam_role_arn')
+        credentials = {}
+
+        for credential_type in ['aws_profile', 'aws_access_key', 'aws_secret_key', 'aws_security_token', 'iam_role_arn']:
+            # This step won't be necessary with https://github.com/ansible/ansible/pull/58288
+            if self.templar.is_template(self.get_option(credential_type)):
+                credentials[credential_type] = self.templar.template(variable=self.get_option(credential_type), disable_lookups=False)
+            else:
+                credentials[credential_type] = self.get_option(credential_type)
+
+        self.boto_profile = credentials['aws_profile']
```

Comment:
```suggestion
        self.boto_profile = credentials.get('aws_profile')
```

Reply:
https://docs.quantifiedcode.com/python-anti-patterns/correctness/not_using_get_to_return_a_default_value_from_a_dictionary.html?highlight=get#use-dict-get-key-default-to-assign-default-values

Reply:
I think the original version is fine. `credentials['aws_profile']` should always be defined. If we use `.get()` here we may actually miss some problem with the block above.